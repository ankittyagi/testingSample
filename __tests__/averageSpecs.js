import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';
import Average from '../components/average';
import { shallow } from 'enzyme';

// On running the test, a snapshot file is created within the __snapshots__ folder
// 'describe' function is used to write a test suite
describe('Average (Snapshot)', () => {
  // 'beforeEach' run before every test
  beforeEach(() => {
    app = shallow(<Average />);
    appy = new Average()
  })

  // 'it' function is used to define a test.
  it('avergae component renders ', () => {
    const component = renderer.create(<Average />);
    const json = component.toJSON();
    // 'expect' function is part of the assertion library exposed by Jest.
    expect(json).toMatchSnapshot();
  });

  it('Method checks', () => {
    expect(app.state().inputValueA).toBeDefined();
    expect(app.state().inputValueA).toBeDefined();
    expect(appy.calculateAverage(12,18)).toBe(15)
  });
});
