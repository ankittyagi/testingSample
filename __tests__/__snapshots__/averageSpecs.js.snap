// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Average (Snapshot) avergae component renders  1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 20,
        "margin": 30,
        "textAlign": "center",
      }
    }
  >
    Average of A and B = 
    0
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Value of A:
  </Text>
  <TextInput
    keyboardType="numeric"
    onChangeText={[Function]}
    placeholder="Value of A"
    style={
      Object {
        "alignSelf": "center",
        "borderColor": "gray",
        "borderWidth": 1,
        "height": 40,
        "lineHeight": 100,
        "marginBottom": 20,
        "textAlign": "center",
        "width": 100,
      }
    }
    value="0"
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Value of B:
  </Text>
  <TextInput
    keyboardType="numeric"
    onChangeText={[Function]}
    placeholder="Value of B"
    style={
      Object {
        "alignSelf": "center",
        "borderColor": "gray",
        "borderWidth": 1,
        "height": 40,
        "lineHeight": 100,
        "marginBottom": 20,
        "textAlign": "center",
        "width": 100,
      }
    }
    value="0"
  />
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#0C42FD",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
            Object {
              "color": "#841584",
            },
          ]
        }
      >
        Calculate Average
      </Text>
    </View>
  </View>
</View>
`;
